content/posts/efficient-navigation-in-vim.md

```diff
@@ -0,0 +1,140 @@
+
+### fzf
+
+One of the most useful plugins is [fzf](https://github.com/junegunn/fzf.vim) which is a great one to have in general. It offers many helpful commands like `:BLines` for searching the current buffer with intelligent fuzzy matching or `:Rg` for even searching the whole project for specific patterns which, in contrast to the other methods we've looked at so far, lets the user navigate between files. fzf is a tool that can be used for jumping between files and buffers but also between different locations within them. It is highly recommended to check it out and add the following lines to the configuration:
+
+```vim
+" Jump to specific file
+nnoremap <C-P> :Files<cr>
+" Search whole project
+nnoremap \ :Rg<space>
+```
+
+The first line lets the user open files in the current project by pressing `ctrl+p` and then typing with fuzzy matching. The second line allows to search the whole project with fuzzy matching by pressing `\` (in contrast to `/` which searches the current buffer).
+
+### EasyMotion
+
+Another useful plugin is [EasyMotion](https://github.com/easymotion/vim-easymotion). Instead of pressing `n` multiple times after a search to jump to the correct occurrence of the search term, it highlights each occurrence and assigns them a unique character combination. By pressing this combination, one can quickly jump to that occurrence with very few key strokes.
+
+An added bonus is that something very similar exists as a browser extension which allows to use many other Vim commands in a web browser and make browsing using only the keyboard possible. It's called [Vimium](https://github.com/philc/vimium).
+
+## Conclusion
+
+Obviously, there are many other ways of navigating Vim buffer(s) that aren't covered here. These are only some of the navigation techniques that can be used and which ones are most convenient and intuitive in which situations is rather subjective.
+
+Another thing to note is that this article talks only about motion commands. It does not discuss commands that go into insert mode while moving the cursor to a specific place (e.g. the `A` command which moves the cursor to the end of a line and enters insert mode). While those can be incredibly useful, discussing them here would have gone beyond the scope of this article.
+
+It is highly recommended to get to know as many different techniques as possible and carefully thinking about which one makes sense in which situation. Users shouldn't be afraid of forcing themselves a bit to use different ones until they have found their own toolbox of useful ones for different situations. It takes some practice to get these into muscle memory but it's worth it all the while to make Vim the efficient text editor it can be.
```

Comment:
Again, I would be a bit more relaxed in using `I highly recommend` and `you` in this paragraph.

`I highly recommend that you get to know`
`You shouldn't be afraid ... until you have found your own`